cpp/src/io/utilities/datasource.cpp

```diff
@@ -41,7 +41,7 @@ class file_source : public datasource {
 
   bool supports_device_read() const override { return _cufile_in != nullptr; }
 
-  bool is_device_read_preferred(size_t size) const
+  bool is_device_read_preferred(size_t size) const override
```

Comment:
Interesting. How did it even work before?

Reply:
It's not an error to omit this. Should've just given a warning.

Reply:
I thought it needs either `virtual` or `override` to avoid overriding the base implementation. Either way, this is a good change.